scripts/build_industrial_energy_demand_per_node.py

```diff
@@ -136,80 +132,149 @@ def create_nodal_electricity_profiles(
 
         # Create weighted profile (8760 hours from reference year)
         node_profile = pd.Series(0.0, index=ffe_profiles.index)
-        total_weight = 0
 
         for sector, demand in sector_demands.items():
-            if demand > 0 and sector in industry_category_to_profile:
-                profile_name = industry_category_to_profile[sector]
-                if profile_name in ffe_profiles.columns:
-                    node_profile += ffe_profiles[profile_name] * demand
-                    total_weight += demand
-                else:
-                    logger.warning(
-                        f"Profile '{profile_name}' not found in FfE data for sector '{sector}'. Using flat demand"
-                    )
-                    flat_profile = pd.Series(
-                        1.0 / len(ffe_profiles), index=ffe_profiles.index
-                    )
-                    node_profile += flat_profile * demand
-                    total_weight += demand
+            profile_name = industry_category_to_profile[sector]
+            if profile_name not in ffe_profiles.columns:
+                raise ValueError(f"Profile '{profile_name}' for sector '{sector}' not in FfE data")
+            node_profile += ffe_profiles[profile_name] * demand
 
         # Map profile to snapshots with correct day-of-week alignment
-        hourly_profile = map_profile_to_snapshots(node_profile, snapshots)
+        hourly_profile = map_profile_to_snapshots(node_profile, snapshots, node_country=node[:2], tol=0.02) # Tolerance increased to 2% to account for holiday replacement and date adjustments
```

Comment:
Instead of `node[:2]`, use `node.map(n.buses.country)` to avoid string inferral.